test/transaction.test.js

```diff
@@ -173,31 +173,37 @@ describe('transactions', function() {
   });
 
   describe('timeout', function() {
+    const TIMEOUT = 50;
     before(function() {
       // Reset the collection
       db.connector.data = {};
     });
 
     var post = {title: 't3', content: 'c3'};
-    before(createPostInTx(post, 50));
+    beforeEach(createPostInTx(post, TIMEOUT));
 
     it('should report timeout', function(done) {
-      setTimeout(function() {
+      // wait until the "create post" transaction times out
+      setTimeout(runTheTest, TIMEOUT * 3);
+
+      function runTheTest() {
         Post.find({where: {title: 't3'}}, {transaction: currentTx},
           function(err, posts) {
-            if (err) return done(err);
-            expect(posts.length).to.be.eql(1);
+            expect(err).to.match(/transaction.*not active/);
             done();
           });
-      }, 300);
-      done();
+      }
     });
 
     it('should invoke the timeout hook', function(done) {
       currentTx.observe('timeout', function(context, next) {
         next();
         done();
       });
+
+      // If the event is not fired quickly enough, then the test can
+      // quickly fail - no need to wait full two seconds (Mocha's default)
+      this.timeout(TIMEOUT * 3);
```

Comment:
nitpick: wouldn't it be set before [test 'should report timeout'](https://github.com/strongloop/loopback-connector/pull/126/files#diff-a9b8fcfc6b0fa14ca33f730f19318021R185)?

Reply:
The timeout function must be called from the test/hook that it is supposed to apply to. See https://mochajs.org/#timeouts

If called before the test (`it` call), it would apply to the entire test suite (`describe` block).